src/node_file_sync.cc

```diff
@@ -0,0 +1,297 @@
+
+  FS_SYNC_TRACE_BEGIN(open);
+  uv_file file = uv_fs_open(nullptr, &req, *path, flags, 438, nullptr);
+  FS_SYNC_TRACE_END(open);
+  if (req.result < 0) {
+    // req will be cleaned up by scope leave.
+    Local<Value> out[] = {
+        Integer::New(isolate, req.result),       // errno
+        FIXED_ONE_BYTE_STRING(isolate, "open"),  // syscall
+    };
+    return args.GetReturnValue().Set(Array::New(isolate, out, arraysize(out)));
+  }
+  uv_fs_req_cleanup(&req);
+
+  auto defer_close = OnScopeLeave([file]() {
+    uv_fs_t close_req;
+    CHECK_EQ(0, uv_fs_close(nullptr, &close_req, file, nullptr));
+    uv_fs_req_cleanup(&close_req);
+  });
+
+  std::string result{};
+  char buffer[8192];
+  uv_buf_t buf = uv_buf_init(buffer, sizeof(buffer));
+
+  FS_SYNC_TRACE_BEGIN(read);
+  while (true) {
+    auto r = uv_fs_read(nullptr, &req, file, &buf, 1, -1, nullptr);
+    if (req.result < 0) {
+      FS_SYNC_TRACE_END(read);
+      // req will be cleaned up by scope leave.
+      Local<Value> out[] = {
+          Integer::New(isolate, req.result),       // errno
+          FIXED_ONE_BYTE_STRING(isolate, "read"),  // syscall
+      };
+      return args.GetReturnValue().Set(
+          Array::New(isolate, out, arraysize(out)));
+    }
+    uv_fs_req_cleanup(&req);
+    if (r <= 0) {
+      break;
+    }
+    result.append(buf.base, r);
+  }
+  FS_SYNC_TRACE_END(read);
+
+  args.GetReturnValue().Set(String::NewFromUtf8(env->isolate(),
+                                                result.data(),
+                                                v8::NewStringType::kNormal,
+                                                result.size())
+                                .ToLocalChecked());
+}
+
+void BindingData::CreatePerIsolateProperties(IsolateData* isolate_data,
+                                             Local<ObjectTemplate> target) {
+  Isolate* isolate = isolate_data->isolate();
+  SetMethodNoSideEffect(isolate, target, "access", Access);
+  SetFastMethodNoSideEffect(isolate, target, "exists", Exists, &fast_exists_);
+  SetMethodNoSideEffect(isolate, target, "readFileUtf8", ReadFileUtf8);
+}
+
+void BindingData::CreatePerContextProperties(Local<Object> target,
+                                             Local<Value> unused,
+                                             Local<Context> context,
+                                             void* priv) {
+  Realm* realm = Realm::GetCurrent(context);
+  realm->AddBindingData<BindingData>(target);
+}
+
+void BindingData::RegisterExternalReferences(
+    ExternalReferenceRegistry* registry) {
+  registry->Register(Access);
+  registry->Register(Exists);
+  registry->Register(FastExists);
+  registry->Register(fast_exists_.GetTypeInfo());
+  registry->Register(ReadFileUtf8);
+}
+
+}  // namespace fs_sync
+
+}  // namespace node
+
+NODE_BINDING_CONTEXT_AWARE_INTERNAL(
+    fs_sync, node::fs_sync::BindingData::CreatePerContextProperties)
```

Comment:
Why does this need to be in its own binding? It seems perfectly fine to just add these to the fs binding..then there is no need to have one more binding data either...

Reply:
Purely for aesthetic purposes. It's extremely hard for me to navigate through the `node_file.cc`. I can move it back to the original file at the end.